src/teachy_tv.c

```diff
@@ -0,0 +1,1775 @@
+    return temp;
+}
+
+void TeachyTvSetSpriteCoordsAndSwitchFrame(u8 objId, u16 x, u16 y, u8 frame)
+{
+    gSprites[objId].pos2.x = x;
+    gSprites[objId].pos2.y = y;
+    gSprites[objId].invisible = 0;
+    StartSpriteAnim(&gSprites[objId], frame);
+}
+
+void TeachyTvSetWindowRegs()
+{
+    SetGpuReg(REG_OFFSET_WIN0V, 0xC64);
+    SetGpuReg(REG_OFFSET_WIN0H, 0x1CD4);
+}
+
+void TeachyTvClearWindowRegs()
+{
+    SetGpuReg(REG_OFFSET_WIN0V, 0x0);
+    SetGpuReg(REG_OFFSET_WIN0H, 0x0);
+}
+
+void TeachyTvBg2AnimController()
+{
+    u16 *tilemapBuffer;
+    u8 i;
+    u32 offset, offset2, j;
+
+    tilemapBuffer = (u16 *)GetBgTilemapBuffer(2);
+    i = 1;
+    do
+    {
+        offset2 = 2;
+        offset = 0x20 * i;
+        j = i + 1;
+        do
+        {
+            tilemapBuffer[offset + offset2] = ((Random() & 3) << 10) + 0x301F;
+            offset2 = (offset2 + 1) << 0x18 >> 0x18;
+        }
+        while ( offset2 <= 0x1B );
+        i = j;
+    }
+    while ( j << 0x18 >> 0x18 <= 0xC );
+    schedule_bg_copy_tilemap_to_vram(2);
+}
+
+void TeachyTvSetupPostBattleWindowAndObj(u8 taskId)
+{
+    u16 *v2 = gTasks[taskId].data;
+    int v1 = ((s16 *)v2)[1];
+    struct Sprite *v3 = &gSprites[v1];
+    int op;
+
+    ClearWindowTilemap(1);
+    TeachyTvClearWindowRegs();
+    op = gTeachyTV_StaticResources.optionChosen;
+    if ( op >= 0 )
+    {
+        if ( op <= 3 )
+        {
+            TeachyTvSetSpriteCoordsAndSwitchFrame(((char*)v2)[2], 0x78, 0x38, 0);
+            ChangeBgX(3, 0x3000, 1);
+            ChangeBgY(3, 0x3000, 2);
+            gUnknown_203F450->var_4004 += 3;
+            gUnknown_203F450->var_4005 -= 3;
+        }
+        else if ( op <= 5 )
+            TeachyTvSetSpriteCoordsAndSwitchFrame(((char*)v2)[2], 0x78, 0x38, 0);
+    }
+
+    v2[4] = 0;
+    v2[5] = 0;
+    TeachyTvGrassAnimationMain(taskId, v3->pos2.x, v3->pos2.y, 0, 1);
+}
+
+void TeachyTvInitTextPrinter(const char *text)
+{
+    u8 spd;
+    gTextFlags.autoScroll = 0;
+    spd = GetTextSpeedSetting();
+    AddTextPrinterParameterized2(0, 4, (const char *)text, spd, 0, 1, 0xC, 3);
+}
+
+void TeachyTvFree()
+{
+    Free(gUnknown_203F450);
+    gUnknown_203F450 = NULL;
+    FreeAllWindowBuffers();
+}
+
+void TeachyTvQuitBeginFade(u8 taskId)
+{
+    BeginNormalPaletteFade(0xFFFFFFFF, 0, 0, 0x10, 0);
+    gTasks[taskId].func = TeachyTvQuitFadeControlAndTaskDel;
+}
+
+void TeachyTvQuitFadeControlAndTaskDel(u8 taskId)
+{
+    if ( !(gPaletteFade.active) )
+    {
+        if ( gUnknown_203F450->state )
+        {
+            SetMainCallback2(*(void (**)())gUnknown_203F450);
+        }
+        else
+        {
+            sub_8055DC4();
+            SetMainCallback2((void (*)())gTeachyTV_StaticResources.callback);
+        }
+        TeachyTvFree();
+        DestroyTask(taskId);
+    }
+}
+
+#ifdef NONMATCHING
+void TeachyTvOptionListController(u8 taskId)
+{
+    char *v2;
+    s32 v3;
+
+    v2 = (char *)gTasks[taskId].data;
```

Comment:
Try making this an `s16 *` variable.  See comment below.

Reply:
still can't match it. I guess we still need to figure out how the control flow is generated